src/components/Visit/visitlist.tsx

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from 'react'
+
+import { useFirestore } from '@/context/firestore'
+import { VisitData } from '@/types/types'
+
+import VisitInstance from './visitinstance'
+
+interface VisitListProps {
+  searchQuery: string
+}
+
+export const VisitList = (props: VisitListProps) => {
+  const { userDoc } = useFirestore()
+  const [visits, setVisits] = useState(userDoc.visits)
+  useEffect(() => {
+    setVisits(userDoc.visits)
+  }, [userDoc.visits])
+
+  const matchesDisplayName = (post: VisitData) =>
+    post.displayName.toLowerCase().includes(props.searchQuery.toLowerCase())
+
+  const matchespetNames = (post: VisitData) =>
+    post.petNames.toLowerCase().includes(props.searchQuery.toLowerCase())
+
+  const matchesSearchTerms = (post: VisitData) =>
+    matchesDisplayName(post) || matchespetNames(post)
+
+  return (
+    <div className=''>
+      {visits
+        .filter((post: VisitData) => {
+          if (props.searchQuery === '' || matchesSearchTerms(post)) {
+            return post
+          }
+        })
```

Comment:
Add a conditional statement to check if the `visits` is undefined. App crashes for new users as they have no visits yet.